client/src/app/components/page/task-detail/task-detail.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router} from '@angular/router';
+import {TaskService} from '../../../services/task.service';
+import {Task} from '../../../models/task';
+import {FormBuilder, FormGroup} from '@angular/forms';
+import {AuthenticationService} from '../../../services/authentication.service';
+import {ModalService} from '../../../services/modal.service';
+import {YesNoModalParams} from '../../../models/modals'
+import {TaskUpdate, TaskAdd} from '../../../models/task'
+@Component({
+  selector: 'app-task-detail',
+  templateUrl: './task-detail.component.html',
+  styleUrls: ['./task-detail.component.scss']
+})
+export class TaskDetailComponent implements OnInit {
+  updateMode:boolean = true;
+  buttonText:string = 'Update';
+  id: string;
+  task:Task;
+  editMode:boolean = false;
+  taskForm:FormGroup;
+  constructor(
+    private route:ActivatedRoute,
+    private ts:TaskService,
+    private fb : FormBuilder, 
+    private as : AuthenticationService,
+    private router : Router,
+    private modalService : ModalService
+    ) { }
+    
+  ngOnInit() {
+    this.id = this.route.snapshot.paramMap.get('id');
+    if(this.id === 'newTask'){
+      this.updateMode=false;
+      this.buttonText='Add task';
+    }
+    this.taskForm = this.fb.group({
+      title:[''],
+      description:['']
+    });
+    this.ts.getTaskById(this.id).subscribe(task=>{
```

Comment:
I think you need to search task data only if task id exist!
if not, than we creating a new one

Reply:
You're right. Changed it